### PR TITLE
Set up split testing mechanism, and first test (Ask CFPB H1s)

### PR DESCRIPTION
**Pull request by Scotchester**
Background: https://GHE/CFGOV/platform/issues/2812

These changes set up Wagtail pages to be able to be "split tested" – i.e., some pages are rendered differently. Our analytics team will track the pages we're experimenting on (grouped into "clusters") and see how the changes perform.

The first test is changing some Ask CFPB answer pages' main headings to use `<h1>` instead of `<h2>`.

## Additions

- A data file defining the clusters as groups of page IDs
  - This includes the first set of clusters, as provided to us by the analytics team.
- A new flag condition that takes the name of a cluster group and a `Page` object and checks to see if that `page.id` is found in that cluster group
- A new feature flag for the first experiment, `ASK_CFPB_H1`, used in the answer page template to set the main heading level to `h1` (rather than the current `h2`) on the answers in the clusters
  - The flag is on by default for answers in the clusters, and if successful, the flag can be given the boolean `True` condition to apply it to all answer pages until the flag is removed and the change is made permanent.
  - There is also an addition to the `AnswerPage` model class that overrides the default thing that the condition checks against the clusters – `page.id` – with the much more common way that we refer to Ask CFPB answers, their `answer.id`.

## Changes

- Removes a few `request` arguments from feature flag checks and the docs, because they are no longer required
- Something in me broke while working on this PR and I couldn't take the `base.py` linting errors in my editor anymore... so I fixed them. One small step towards un-ignoring our `settings` folder in flake8!
  - If you want to ignore the linting changes, the only substantive-to-this-PR change in `base.py` is the addition of the `ASK_CFPB_H1` flag (new lines 686-691).

## Testing

1. Pull branch
1. Visit http://localhost:8000/ask-cfpb/what-is-negative-amortization-en-103/ and see a larger main heading
   - Inspect it and see that it's an `h1`
1. Visit an answer that's not in the clusters, like http://localhost:8000/ask-cfpb/how-can-i-figure-out-if-i-can-afford-to-buy-a-home-and-take-out-a-mortgage-en-118/, and see that it's still rendering as an `h2`
1. Run the tests with `tox -e fast ask_cfpb.tests.models.test_pages` and `tox -e fast core.tests.test_feature_flags`

## Notes

- I don't remember if I was supposed to set the `h1` we're testing to display at the same size as `h2` or not. @kelleyholden, do you remember what we discussed?

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [x] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [x] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

**Comment by Scotchester** [comment]
D'oh! Didn't pull from the branch before rebasing on `master` 🤦‍♂️

**Comment by Scotchester** [comment]
I have updated this PR to generalize it (make it not specific to Ask CFPB). Thanks to @chosak for nudging me in the right direction. I have also updated the original post accordingly.

Ready for re-review!

**Comment by Scotchester** [comment]
@chosak Correctly (privately) rebuked me for not including any documentation in this PR :) On hold again while I find time to write that.

**Comment by Scotchester** [comment]
@chosak I haven't dealt with your organizational suggestions yet, but here's a draft of the content that I would love your feedback on: https://github.com/cfpb/cfgov-refresh/blob/split-testing/docs/split-testing.md

File: docs/split-testing.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+In contrast to traditional A/B testing,
+where the change being tested is shown to a portion of the _audience_,
+in the case of split testing,
+the change is shown to everyone, but only on a subset of similar pages
+(e.g., a group of related Ask CFPB answers).
+
+
+## So you've been asked to set up a split testing experiment
+
+Good news! The infrastructure has been put in place for you to do it easily.
```
Comment (willbarton) [inline docs/split-testing.md]: Good news, everyone!

File: docs/split-testing.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+It's set up so that the code being tested is isolated by a feature flag.
+
+### The flag condition
+
+In order to check whether or not a page is part of a split testing cluster,
+we registered a new flag condition in `cfgov/core/feature_flags.py`.
```
Comment (willbarton) [inline docs/split-testing.md]: Is it worth linking "flag condition" to the Django-Flags docs? Maybe [the concepts overview](https://cfpb.github.io/django-flags/#concepts)? I'm not sure there's a good link target (something to fix in Django-Flags!), but "registered a new flag condition" may not be terribly helpful for someone who's been told to implement split testing but doesn't know how our flags library works.

File: docs/split-testing.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+
+## So you've been asked to set up a split testing experiment
+
+Good news! The infrastructure has been put in place for you to do it easily.
+It's set up so that the code being tested is isolated by a feature flag.
+
+### The flag condition
+
+In order to check whether or not a page is part of a split testing cluster,
```
Comment (willbarton) [inline docs/split-testing.md]: > split testing cluster

Should what this is be defined? May just as one extra sentence around [line ten](https://github.com/cfpb/cfgov-refresh/pull/4543/files#diff-767bdb4cd43ee4726bf36b549ebe30deR10):

> the change is shown to everyone, but only on a subset of similar pages
> (e.g., a group of related Ask CFPB answers). *We call this a subset of 
> pages a cluster.*

or something like.

File: docs/split-testing.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+we registered a new flag condition in `cfgov/core/feature_flags.py`.
+
+```python
+@conditions.register('in split testing cluster')
+def in_split_testing_cluster(cluster_group, page, clusters=CLUSTERS, **kwargs):
+    cluster_group = clusters[cluster_group]
+    lookup_value = getattr(page, 'split_test_id', page.id)
+    return lookup_value in chain(*cluster_group.values())
+```
```
Comment (willbarton) [inline docs/split-testing.md]: Is it necessary to include the full code for the condition if someone who is setting up a split testing cluster doesn't need to modify it? They just need to know it exists and how to use it, right?

File: docs/split-testing.md (hunk 2 above)
Comment (Scotchester) [reply docs/split-testing.md]: Yes! I meant to link to the django-flags docs and forgot. Thanks for the reminder.

Comment (Scotchester) [reply docs/split-testing.md]: Updated in 3b9fcf49384c1515decb1204770ed90def95e859

File: docs/split-testing.md (hunk 3 above)
Comment (Scotchester) [reply docs/split-testing.md]: Updated in 3b9fcf49384c1515decb1204770ed90def95e859

**Comment by Scotchester** [comment]
@chosak D'oh! Yes, I did. Pushed a change to add it to the docs menu.

**Pull request merged**